packages/injective-cosmwasm/src/exchange/types.rs

```diff
@@ -218,7 +219,20 @@ impl ShortSubaccountId {
     {
         let id = id_s.into();
 
-        match id.parse::<u16>() {
+        // let's check first if it's a hexadecimal number and parse to decimal (expected by the chain)
+        let as_decimal = match u32::from_str_radix(id.as_str(), 16) {
+            Ok(dec) => Ok(dec),
+            Err(_) => Err(StdError::generic_err("Invalid value: ShortSubaccountId was not a hexadecimal number")),
+        };
+
+        // if the string cannot be parsed to a decimal return error
+        let decimal_u32 = as_decimal?;
+
+        // if it is a hexadecimal number, let's pad it with left zeros
+        let number_to_use = format!("{:0>3}", decimal_u32.to_string());
+
+        // check if resulting number isn't bigger than chain limit
+        match number_to_use.parse::<u16>() {
```

Comment:
I don't think we need this, do we? if we already do the conversion to decimal during serializations.

- If we do, please say why and remove the comments.
- If we don't, please remove here.